fv3gfs/util/buffer.py

```diff
@@ -91,4 +92,4 @@ def recv_buffer(allocator: Callable, array: ndarray, timer: Optional[Timer] = No
             timer.stop("unpack")
             yield recvbuf
             with timer.clock("unpack"):
-                array[:] = recvbuf
+                array[:] = cp.asarray(recvbuf) if isinstance(array, cp.ndarray) else recvbuf
```

Comment:
Please add a code comment explaining why this was important, edit as needed.
```suggestion
                # without cp.asarray, it throws a CUDA error about
                # memory ownership on the cupy arrays from the gt4py storages
                # this should be fixed in a later version of gt4py
                array[:] = cp.asarray(recvbuf) if isinstance(array, cp.ndarray) else recvbuf
```